cmd/osde2ectl/get/cmd.go

```diff
@@ -99,28 +132,67 @@ func run(cmd *cobra.Command, argv []string) error {
 	if properties := cluster.Properties(); properties["MadeByOSDe2e"] != "true" {
 		return fmt.Errorf("Cluster was not created by osde2e")
 	}
-	log.Printf("Cluster name - %s and Cluster ID - %s", cluster.ID(), cluster.Name())
+	log.Printf("Cluster name - %s and Cluster ID - %s", cluster.Name(), cluster.ID())
 
 	if kubeconfigStatus {
-		err := setKubeconfig(clusterID)
+		content, err := setKubeconfig(clusterID)
 		if err != nil {
 			return fmt.Errorf("Error getting the cluster's kubeconfig - %s", err)
 		}
+
+		filename := cluster.Name() + "-kubeconfig.txt"
+
+		var filePath string
+		if viper.GetString(config.Kubeconfig.Path) != "" {
+			_, err := os.Stat(viper.GetString(config.Kubeconfig.Path))
+			if err != nil {
+				fmt.Println("Path directory is invalid - ", err.Error(), "Will create a new directory.")
+				err = os.Mkdir(viper.GetString(config.Kubeconfig.Path), os.ModePerm)
+				if err != nil {
+					return fmt.Errorf("Unable to create a new directory - %s", err.Error())
+				}
+			}
+			filePath = filepath.Join(viper.GetString(config.Kubeconfig.Path), filename)
+		} else {
+			dir, err := os.Getwd()
+			if err != nil {
+				return fmt.Errorf("Unable to get CWD: %s", err.Error())
+			}
+
+			filePath = filepath.Join(viper.GetString(dir), filename)
+		}
+		err = ioutil.WriteFile(filePath, content, os.ModePerm)
+
+		if err != nil {
+			return fmt.Errorf("could not write KubeConfig into a file: %v", err)
+		}
+		fmt.Println("Successfully downloaded the kubeconfig into -", filePath, ". Run the command \"export TEST_KUBECONFIG=", filePath, "\"")
+		if args.hours == 0 && args.minutes == 0 && args.seconds == 0 {
+			if err = provider.ExtendExpiry(clusterID, 0, 30, 0); err != nil {
```

Comment:
Does this effectively state that if we try to get the kubeconfig, we should extend the expiry by 30 minutes? I like the idea, but let's make a check to only do this if the cluster is within 30 minutes of being terminated.